src/Razor/Razor.Runtime/src/RazorRazorPageBase.cs

```diff
@@ -0,0 +1,505 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Razor.Runtime.TagHelpers;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+
+namespace Microsoft.AspNetCore.Razor
+{
+    /// <summary>
+    /// Represents properties and methods that are needed in order to render a view that uses Razor syntax.
+    /// </summary>
+    public abstract class RazorRazorPageBase<TOutputContext> : IRazorRazorPage<TOutputContext> where TOutputContext : IOutputContext
+    {
+        private readonly Stack<TextWriter> _textWriterStack = new Stack<TextWriter>();
+        private StringWriter _valueBuffer;
+        private AttributeInfo _attributeInfo;
+        private TagHelperAttributeInfo _tagHelperAttributeInfo;
+
+        public virtual TOutputContext ViewContext { get; set; }
```

Comment:
I feel kinda iffy about putting a context type here, and it would really be preferrable to make this based on inject instead. 

Did you have particular scenarios in mind for this other than the fact that MVC has a context?

Reply:
Only had getting-rid-of-mvc in mind here.. We'd need at least an interface for it to get it in via [Inject], though.